src/main.c

```diff
@@ -476,11 +480,19 @@ static char* jsonToString(Json *json) {
   }
 }
 
-static bool is_valid_key(char *key) {
+// Zowe.environments key must follow Unix variable name syntax:
+// alphaNum | underscore & first char is not a digit
```

Comment:
The precedence of the operators is not very obvious. Could you add parenthesis?

`(alphaNum | underscore) & first char is not a digit`

Or just write it like this:
```
// Zowe.environments key must follow Unix variable name syntax:
// * The first char must not be a digit
// * Any characters must be either alphanumeric or an underscore
```